src/Nethermind/Nethermind.Consensus.Ethash/DifficultyCalculator.cs

```diff
@@ -39,12 +39,12 @@ public DifficultyCalculator(ISpecProvider specProvider)
             bool parentHasUncles)
         {
             IReleaseSpec spec = _specProvider.GetSpec(blockNumber);
-            BigInteger baseIncrease = BigInteger.Divide(parentDifficulty, spec.DifficultyBoundDivisor);
-            BigInteger timeAdjustment = TimeAdjustment(spec, parentTimestamp, currentTimestamp, parentHasUncles);
+            BigInteger baseIncrease = BigInteger.Divide((BigInteger)parentDifficulty, spec.DifficultyBoundDivisor);
```

Comment:
still BigInteger?

Reply:
yeah, too much hassle here with negative values that need to be Int256 and then applied in operation sto UInt256 - lots of conversions / operators are missing so leaving it here